connect/eaas/runner/managers/transformation.py

```diff
@@ -0,0 +1,282 @@
+            )
+
+        return self.client
+
+    async def get_argument(self, task_data):
+        """
+        Get the transformation request through Connect public API
+        related to the processing task.
+        """
+        client = self.get_client(task_data)
+        object_exists = await self.filter_collection_by_event_definition(
+            client,
+            task_data,
+        )
+        if not object_exists:
+            return
+
+        definition = self.config.event_definitions[task_data.input.event_type]
+        url = definition.api_resource_endpoint.format(pk=task_data.input.object_id)
+        resource = AsyncResource(client, url)
+
+        return await resource.get()
+
+    def get_method_name(self, task_data, argument):
+        method_name = task_data.input.data['method_name']
+        if not task_data.input.data['status']:
+            return None
+
+        return method_name
+
+    async def invoke(self, task_data, method, tfn_request):
+        """
+        Creates async task for transformation process.
+
+        :param task_data: Data of the task to be processed.
+        :type task_data: connect.eaas.core.proto.Task
+        :param method: The method that has to be invoked.
+        :param tfn_request: Connect tfn request that need to be processed.
+        :type tfn_request: dict
+        """
+
+        future = asyncio.create_task(self.process_transformation(
+            task_data, tfn_request, method,
+        ))
+
+        logger.info(f'Enqueue result for task {task_data.options.task_id}')
+        asyncio.create_task(self.enqueue_result(task_data, future))
+
+    async def build_response(self, task_data, future):
+        result_message = Task(**task_data.dict())
+        try:
+            begin_ts = time.monotonic()
+            result = await asyncio.wait_for(
+                future,
+                timeout=self.config.get_timeout('transformation'),
+            )
+            result_message.output = TaskOutput(result=result.status)
+            result_message.output.runtime = time.monotonic() - begin_ts
+            logger.info(
+                f'Transformation task {task_data.options.task_id} '
+                f'result: {result.status}, '
+                f'took: {result_message.output.runtime}',
+            )
+            if result.status in (ResultType.SKIP, ResultType.FAIL):
+                result_message.output.message = result.output
+        except Exception as e:
+            self.log_exception(task_data, e)
+            result_message.output = TaskOutput(result=ResultType.RETRY)
+            result_message.output.message = traceback.format_exc()[:4000]
+
+        return result_message
+
+    async def process_transformation(self, task_data, tfn_request, method):
+        download_result = await asyncio.get_running_loop().run_in_executor(
+            self.executor,
+            self.download_exel,
+            tfn_request,
+        )
+        input_file = download_result.reslut()
```

Comment:
Since you already awaited the future (asyncio.Future) returned by `run_in_executor`, `download_result` already is your NamedTemporaryFile.